### PR TITLE
Add known-bug for #95034

### PR DESCRIPTION
**Pull request by compiler-errors**
Couldn't fix the issue, since I am no type theorist and inference variables in universes above U0 scare me. But I at least wanted to add a known-bug test for it.

cc #95034 (does not fix)

### CODE REVIEW

File: src/test/ui/hrtb/issue-94034.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
+// known-bug
```
Comment (compiler-errors) [inline src/test/ui/hrtb/issue-94034.rs:1]: I might take all of these and turn them into like... `known-ice` or something. Like automatically expect failure status 101, set backtrace=0, normalize all of the ICE notes, etc.

Comment (jackh726) [reply src/test/ui/hrtb/issue-94034.rs:1]: That would be nice :)

**Comment by compiler-errors** [comment]
I added more test normalization, to account for wasm32 target not ICEing in the same place...

**Pull request merged**